components/plan-build/bin/public.bash

```diff
@@ -367,8 +369,10 @@ download_file() {
 # will be printed to stderr with the expected and computed shasum values.
 verify_file() {
   build_line "Verifying $1"
-  local checksum=($($_shasum_cmd $HAB_CACHE_SRC_PATH/$1))
-  if [[ $2 = $checksum ]]; then
+  local checksum
+  # shellcheck disable=2154
+  read -r checksum _ < <($_shasum_cmd "$HAB_CACHE_SRC_PATH"/"$1")
+  if [[ $2 = "$checksum" ]]; then
```

Comment:
Previously, `checksum` was an array, but treated as a scalar in the expression
```bash
[[ $2 = $checksum ]]
```
This technically works, because `$checksum` will implicitly access the first element (as opposed to `${checksm[0]}"` which does it explicity). This is confusing and obscures the fact that `checksum` is an array variable.

Instead, I changed the assignment of `checksum` to be a scalar and explicitly ignore the rest of the output of `$_shasum_cmd`.

Reply:
nice catch